src/index.js

```diff
@@ -0,0 +1,120 @@
+let defaultLogger = console;
+
+/**
+  @class Catchie
+  Consume errors and retry the function execution given times
+
+  Supported environment variables:
+  MAX_REPEAT - cap amount of function retry
+**/
+class Catchie {
+  /**
+    @constructor
+    Construct instance of the module
+    @param logger {Object} - logger instnace
+    @param silent {boolean} - log or not the info about retries
+  **/
+  constructor(logger, silent) {
+    this.logger = logger || defaultLogger;
+    this.successCount = 0;
+    this.failCount = 0;
+    this.silent = silent;
+  }
+
+  /**
+    @function clear
+    Reset the values of successCount and failureCount
+  **/
+  clear() {
+    this.successCount = 0;
+    this.failureCount = 0;
+  }
+
+  /**
+    @function callCount
+    Get the total amount of given function calls
+    @return {number} - total amount of function calls
+  **/
+  get callCount() {
+    return this.successCount + this.failureCount;
+  }
+
+  /**
+    @function retry
+    Execute the function given amount of times
+    @param fn { Function } - a function returning value / Promise
+    @retryCount { string|number } - amount of times to repeat a function on error
+    @return
+      { any } - if function is sync
+      { Promise } - if function is async
+    @throws if after the retryCount times of function execution the function still ended up with an error
+  **/
+  retry(fn, retryCount = 1) {
+    this.clear();
+    let timesToRetry = parseInt(retryCount, 10);
+
+    if (isNaN(timesToRetry)) {
+      timesToRetry = 0;
+    }
+
+    if (typeof fn !== 'function') {
+      throw new Error('Wrong function type');
+    }
+
+    let timesRetried = 0;
+
+    const loopCheck = (e, args) => {
+      if (timesRetried >= timesToRetry) {
+        throw new Error(e);
+      }
+
+      timesRetried += 1;
+      if (!this.silent) {
+        this.logger.info(`Retrying function due to ${e} error`);
+      }
+      return repeatable(...args); // eslint-disable-line
```

Comment:
We had that issue before and unfortunately it can not be fixed because repeatable calles ``loopCheck`` and ``loopCheck`` itself then calls ``repeatable``, so it is either one of them that will cause this error.